### PR TITLE
Fix library dependency checking for Intel targets

### PR DESCRIPTION
**Pull request by kjokinie**
V2: Added comment about grepping the libraries after objdump

The sb2 targets for i486/i586 perform really funny library path
translations when the binaries are run inside sb2. Hence the old
way to detect library dependencies by running them with
LD_TRACE_LOADED_OBJECTS=1 flag does not work anymore correctly.

Fixed by getting direct dependencies of the executables via objdump
and then using ldd to collect dependencies of the direct dependencies.
This way the executable does not need to be run to find the dependencies.

Signed-off-by: Kalle Jokiniemi <kalle.jokiniemi@jolla.com>

### CODE REVIEW

File: initfs/scripts/moslo-build.sh [hunk 1]
```diff
@@ -108,17 +134,26 @@ add_dependencies()
...
                 if [ "$?" -ne "2" ] ; then
-                        LD_LINUX=""
+                        #Get direct dependencies
+                        DIRECT_LIBS=$(objdump_find_lib_path $i)
+                        DEP="$DIRECT_LIBS"
+                        LD_LINUX="ldd"
+                        #Get full dependencies of the direct dependencies
+                        for lib in $DIRECT_LIBS; do
+                                TEMPDEPS=$($LD_LINUX $lib \
+                                        | sed -ne "s/.*[\t ]\(\/.*\) (.*/\1/gp")
```
Comment (deztructor) [inline initfs/scripts/moslo-build.sh]: A bit more readable to use ```sed -ne "s|.*[\t ]\(/.*\) (.*|\1|gp"```, one backslash less :)

File: initfs/scripts/moslo-build.sh [hunk 2]
```diff
@@ -108,17 +134,26 @@ add_dependencies()
...
+                                DEP="$DEP $TEMPDEPS"
+                        done
                 else
-                        LD_LINUX=$LD
+                        #Get dependencies
+                        DEP=$($LD_LINUX $i \
+                                | sed -ne "s/.*[\t ]\(\/.*\) (.*/\1/gp")
```
Comment (deztructor) [inline initfs/scripts/moslo-build.sh]: Share regexp or even whole expression in the variable/function? You are using it in both branches of if/else.

File: initfs/scripts/moslo-build.sh [hunk 3]
```diff
@@ -58,6 +58,32 @@ debug()
...
+        local FOUND_LIB=""
+
+        if test -z $1; then
+                echo "find_lib_path: Error: Empty executable name!"
+                exit 1
+        fi
+        LIB_LIST=$(objdump -p "$1" | grep NEEDED | sed -e 's/\NEEDED//g')
+        for lib in $LIB_LIST; do
+                for path in $LIB_PATHS; do
+                        FOUND_LIB=$(find $path/ | grep $lib)
```
Comment (deztructor) [inline initfs/scripts/moslo-build.sh]: I wonder is there a small probability that some library name includes grepped name as a part? e.g. something like libX.so.Y and libglibX.so.Y? Maybe it is better to trim and compare using equal or to use "basename"?

File: initfs/scripts/moslo-build.sh (hunk 1 above)
Comment (kjokinie) [inline initfs/scripts/moslo-build.sh]: For me most sed and awk lines are completely incomprehensible no matter how you write it :P

Was the replacement of "/" with pipe "|" intentional?

File: initfs/scripts/moslo-build.sh (hunk 2 above)
Comment (kjokinie) [inline initfs/scripts/moslo-build.sh]: Sounds reasonable idea. well named function would actually tell what it does. You never really know that from looking at the sed line. Now the problem is just finding out what it does, as I copied it from the old code :P 

I'll fix something for it.

File: initfs/scripts/moslo-build.sh (hunk 3 above)
Comment (kjokinie) [inline initfs/scripts/moslo-build.sh]: hmm.. I originally had it just "find $path/$lib" but it then occured to me that there might be some libraries under sub-directories... I'll see if I can do something there, thanks.

File: initfs/scripts/moslo-build.sh (hunk 2 above)
Comment (kjokinie) [inline initfs/scripts/moslo-build.sh]: Sorry, I can't change it. Passing strings between functions is not so fun in scripts, and if I use variable the quotes become the problem.

File: initfs/scripts/moslo-build.sh (hunk 3 above)
Comment (kjokinie) [inline initfs/scripts/moslo-build.sh]: Fun.. It turns out the way I wrote was actually the thing that made it work by accident. It turns out objdump -p | grep NEEDED points to library symlinks only, and by grepping like I did, I get both the symlink and the actual library listed. Unfortunately it seems the order in which the library and symlink get in from "find" is random, and basename will not always match, even though the libraries are in fact correct. (comparison between symlink and library name is false, i.e libblkid.so.1 vs. libblkid.so.1.1.0).

I will just add a comment there.

**Pull request merged**